sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -509,6 +512,23 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   }
 
   /**
+   * Add a Hint to a logical plan.
+   */
+  private def withHints(
+      ctx: HintContext,
+      relation: LogicalPlan): LogicalPlan = withOrigin(ctx) {
+    val stmt = ctx.hintStatement
+    val name = stmt.hintName.getText.toUpperCase
+    if (stmt.identifierList != null) {
+      Hint(name, visitIdentifierList(stmt.identifierList), relation)
+    } else if (stmt.parameters != null) {
+      Hint(name, stmt.parameters.asScala.map(_.getText), relation)
+    } else {
+      Hint(name, Seq.empty[String], relation)
```

Comment:
I am almost sure this cannot happen... I don't think `stmt.parameters` can be `null`.